nautobot/extras/utils.py

```diff
@@ -615,82 +613,49 @@ def refresh_job_model_from_job_class(job_model_class, job_class, job_queue_class
-        logger.info(f"Pod {pod_name} does not exist in namespace {pod_namespace}. Creating it...")
-        pod_manifest = {
-            "apiVersion": "v1",
-            "kind": "Pod",
-            "metadata": {"name": pod_name},
-            "spec": {
-                "containers": [
-                    {
-                        "image": nautobot_image_name,
-                        "name": nautobot_container_name,
-                    }
-                ]
-            },
-        }
-        # import yaml
-        # with open('./development/pod.yaml', 'r') as f:
-        #     pod_manifest = yaml.safe_load(f)
-        logger.info(f"Creating pod {pod_name} in namespace {pod_namespace}")
-        resp = api_instance.create_namespaced_pod(body=pod_manifest, namespace=pod_namespace)
-        logger.info(f"Reading pod {pod_name} in namespace {pod_namespace}")
-        resp = api_instance.read_namespaced_pod(name=pod_name, namespace=pod_namespace)
-        logger.info("Done.")
-
-    # Calling exec interactively
-    logger.info(f"Waiting for container {nautobot_container_name} to be fully set up")
-    exec_command = ["/bin/sh"]
-    resp = stream(
-        api_instance.connect_get_namespaced_pod_exec,
-        pod_name,
-        pod_namespace,
-        command=exec_command,
-        stderr=True,
-        stdin=True,
-        stdout=True,
-        tty=False,
-        _preload_content=False,
-    )
-    resp.write_stdin("nautobot-server runjob --local -u admin nautobot.core.jobs.ExportObjectList\n")
-    sresult = resp.read_stdout()
-    serror = resp.read_stderr()
-    logger.info(f"Job Result is: {sresult}")
-    logger.info(f"Error is: {serror}")
-    resp.close()
+    configuration = kubernetes.client.Configuration()
+    # configure API Key authorization: BearerToken
+    configuration.host = settings.KUBERNETES_JOB_POD_HOST
+    configuration.ssl_ca_cert = "/var/run/secrets/kubernetes.io/serviceaccount/ca.crt"
+    with open("/var/run/secrets/kubernetes.io/serviceaccount/token", "r") as token_file:
+        token = token_file.read().strip()
+    configuration.api_key_prefix["authorization"] = "Bearer"
+    configuration.api_key["authorization"] = token
+    with kubernetes.client.ApiClient(configuration) as api_client:
+        api_instance = kubernetes.client.BatchV1Api(api_client)
+
+    with open("./development/nautobot-job-job.yaml", "r") as f:
+        pod_manifest = yaml.safe_load(f)
+
+    user = job_result.user.username
+    job_model = job_result.job_model
+    job_class_path = f"{job_model.module_name}.{job_model.job_class_name}"
+    pod_manifest["metadata"]["name"] = "nautobot-job"
+    pod_manifest["spec"]["template"]["spec"]["containers"][0]["command"] = [
+        "nautobot-server",
+        "runjob",
+        "--local",
+        "-u",
+        f"{user}",  # TODO replace these arguments with actual
+        f"{job_class_path}",
+        "-d",
+        f"{job_kwargs}",
+        "-r",
+        f"{job_result.pk}",
+    ]
```

Comment:
If we are passing the job result, we shouldn't need to pass anything else, no?

Reply:
[f1109ec](https://github.com/nautobot/nautobot/pull/6471/commits/f1109ec424f5b1077d2ee298b5bb83a08db1d728) In this commit here, with the new management command we are only passing in the `job_kwargs` and job result.  Correct me if I am wrong but `job_kwargs` seems to be never stored on the job result, it is passed into `apply_async()` by itself.